code/game/gamemodes/dynamic/antag_rulesets.dm

```diff
@@ -0,0 +1,171 @@
+#define MINDSHIELDED_JOBS list("Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Blueshield", "Nanotrasen Representative", "Magistrate", "Internal Affairs Agent", "Nanotrasen Navy Officer", "Special Operations Officer", "Trans-Solar Federation General")
+
+/datum/ruleset
+	/// What this ruleset is called
+	var/name = "BASE RULESET"
+	/// The cost to roll this ruleset
+	var/ruleset_cost = 1
+	/// The weight to roll this ruleset
+	var/ruleset_weight = 1
+	/// The cost to roll an antagonist of this ruleset
+	var/cost = 1
+	/// The weight to roll an antagonist of this ruleset
+	var/weight = 1
+	/// Antagonist datum to apply to users
+	var/datum/antagonist/antagonist
+	/// A ruleset to be added when this ruleset is selected by the gamemode
+	var/datum/ruleset/implied/implied_ruleset
+
+	/// These roles 100% cannot be this antagonist
+	var/list/banned_jobs = list("Cyborg")
+	/// These roles can't be antagonists because mindshielding (this can be disabled via config)
+	var/list/protected_jobs = MINDSHIELDED_JOBS
+	/// Applies the mind roll to assigned_role, preventing them from rolling a normal job. Good for wizards and nuclear operatives.
+	var/assign_job_role = FALSE
+	/// A blacklist of species names that cannot play this antagonist
+	var/list/banned_species = list()
+	/// If true, the species blacklist is now a species whitelist
+	var/banned_species_only = FALSE
+
+	// var/list/banned_mutual_rulesets = list() // UNIMPLEMENTED: could be used to prevent nukies rolling while theres cultists, or wizards, etc
+
+	/* This stuff changes, all stuff above is static */
+	/// How many antagonists to spawn
+	var/antag_amount = 0
+	/// All of the minds that we will make into our antagonist type
+	var/list/datum/mind/pre_antags = list()
+
+/datum/ruleset/proc/ruleset_possible(ruleset_budget, rulesets)
+	if(ruleset_budget < ruleset_cost)
+		return "Not enough budget"
+	if(!length(SSticker.mode.get_players_for_role(antagonist::job_rank))) // this specifically needs to be job_rank not special_rank
+		return "No drafted players"
```

Comment:
This should probably return a constant value that then gets translated to a human readable value when displayed in logs/UI